aten/src/ATen/templates/Functions.h

```diff
@@ -22,6 +22,25 @@ using native::tensor;
 
 ${function_declarations}
 
+// Special C++ only overloads for std()-like functions (See gh-40287)
+// These are needed because int -> bool conversion takes precedence over int -> IntArrayRef
+// So, for example std(0) would select the std(unbiased=False) overload
+inline Tensor var(const Tensor& self, int dim) {
+  return at::native::var(self, IntArrayRef{dim});
+}
+
+inline std::tuple<Tensor,Tensor> var_mean(const Tensor& self, int dim) {
```

Comment:
Should these be `int64_t`?

Reply:
`std(0)` will only work if the type is an exact match for operator overloading, so `dim` must be exactly `int`. I could add more overloads to accept other integral types, but an `int` overload would still be needed.

Reply:
Would be good to explicitly mention this in the comment!